ibc/lightclients/groth16/client_state.go

```diff
@@ -297,59 +300,41 @@ func (cs *ClientState) UpdateState(ctx sdktypes.Context, cdc codec.BinaryCodec,
 	if !ok {
 		return []exported.Height{}, fmt.Errorf("the only supported clientMsg type is Header")
 	}
+
 	height, ok := header.GetHeight().(clienttypes.Height)
 	if !ok {
 		return []exported.Height{}, fmt.Errorf("invalid height type %T", header.GetHeight())
 	}
 
-	// Check if this is a mock proof (all zeros)
-	isMockProof := bytes.Count(header.StateTransitionProof, []byte{0}) == len(header.StateTransitionProof)
-
-	// If this is a mock proof, we don't need to verify it.
-	if !isMockProof {
-
-		// This is a real proof, so we need to verify it.
-		trustedConsensusState, err := GetConsensusState(clientStore, cdc, clienttypes.NewHeight(0, uint64(header.TrustedHeight)))
-		if err != nil {
-			return []exported.Height{}, fmt.Errorf("failed to get trusted consensus state: %w", err)
-		}
-		vk, err := DeserializeVerifyingKey(cs.StateTransitionVerifierKey)
-		if err != nil {
-			return []exported.Height{}, fmt.Errorf("failed to deserialize verifying key: %w", err)
-		}
-
-		publicWitness := PublicWitness{
-			TrustedHeight:             header.TrustedHeight,
-			TrustedCelestiaHeaderHash: header.TrustedCelestiaHeaderHash,
-			TrustedRollupStateRoot:    trustedConsensusState.StateRoot,
-			NewHeight:                 header.NewHeight,
-			NewRollupStateRoot:        header.NewStateRoot,
-			NewCelestiaHeaderHash:     header.NewCelestiaHeaderHash,
-			CodeCommitment:            cs.CodeCommitment,
-			GenesisStateRoot:          cs.GenesisStateRoot,
-		}
-
-		witness, err := publicWitness.Generate()
-		if err != nil {
-			return []exported.Height{}, fmt.Errorf("failed to generate state transition public witness: %w", err)
-		}
-
-		proof := groth16.NewProof(ecc.BN254)
-		_, err = proof.ReadFrom(bytes.NewReader(header.StateTransitionProof))
-		if err != nil {
-			return []exported.Height{}, fmt.Errorf("failed to read proof: %w", err)
-		}
-
-		fmt.Printf("Verifying state transition proof...\n")
-		err = groth16.Verify(proof, vk, witness)
-		if err != nil {
-			return []exported.Height{}, fmt.Errorf("failed to verify proof: %w", err)
-		}
+	evmProverRPC := os.Getenv("EVM_PROVER_URL")
+	conn, err := grpc.NewClient(evmProverRPC, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to prover: %w", err)
+	}
+	defer conn.Close()
+	client := proverclient.NewProverClient(conn)
+
+	verifyProofRequest := &proverclient.VerifyProofRequest{
+		Proof:           header.StateTransitionProof,
+		Sp1PublicInputs: header.PublicValues,
+		Sp1VkeyHash:     cs.StateTransitionVerifierKey,
+		Groth16Vk:       cs.Groth16Vk,
+	}
+
+	fmt.Printf("Verifying groth16 state transition proof from height: %d to height: %d\n", header.GetHeight(), header.NewestHeight)
+	verifyProofResponse, err := client.VerifyProof(ctx, verifyProofRequest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to verify proof: %w", err)
+	}
+	fmt.Println(verifyProofResponse, "verify proof response")
```

Comment:
[nit] can this instead do

```suggestion
	fmt.Printf("verify proof response: %v", verifyProofResponse)
```

the current order is confusing.